### PR TITLE
[doc] document missing serve public APIs

### PR DESCRIPTION
**Pull request by can-anyscale**
- Add missing serve public APIs
- Turn on the api policy check for serve now that documentation is complete

Test:
- CI

### CODE REVIEW

File: doc/source/serve/api/index.md [hunk 1]
```diff
@@ -6,7 +6,7 @@
 (core-apis)=
 
 ```{eval-rst}
-.. module:: ray
+.. currentmodule:: ray
```
Comment (can-anyscale) [inline doc/source/serve/api/index.md:9]: update this to be consistent with other docs

**Comment by can-anyscale** [comment]
@edoakes , @shrekris-anyscale let me know if these documentation makes sense, thankks

**Review by shrekris-anyscale — APPROVED** [review]
This looks good to me. The docs are still building, so I haven't seen their formatting yet. @edoakes can you take a look too?

**Review by aslonnie — APPROVED** [review]

**Comment by can-anyscale** [comment]
@shrekris-anyscale , @edoakes the doc finished building now https://anyscale-ray--46819.com.readthedocs.build/en/46819/serve/api/index.html, thankks

**Comment by shrekris-anyscale** [comment]
Is there a reason [`ServeActorDetails`](https://anyscale-ray--46819.com.readthedocs.build/en/46819/serve/api/doc/ray.serve.schema.ServeActorDetails.html#ray.serve.schema.ServeActorDetails), [`ProxyDetails`](https://anyscale-ray--46819.com.readthedocs.build/en/46819/serve/api/doc/ray.serve.schema.ProxyDetails.html#ray.serve.schema.ProxyDetails), and [`EncodingType`](https://anyscale-ray--46819.com.readthedocs.build/en/46819/serve/api/doc/ray.serve.schema.EncodingType.html#ray.serve.schema.EncodingType) are all empty?

`ServeActorDetails` and `ProxyDetails` both lack descriptions also:
Screenshot 2024-07-26 at 2 01 35 PM

**Comment by can-anyscale** [comment]
@shrekris-anyscale good catch, they don't have an existing docstring https://github.com/ray-project/ray/blob/master/python/ray/serve/schema.py, i can add a description for them in this PR, let me know

**Comment by shrekris-anyscale** [comment]
Sure that would be helpful. How about:

* `ServeActorDetails`:

```python
"""Detailed info about a Ray Serve actor."""
```

* `ProxyDetails`:

```
"""Detailed info about a Ray Serve ProxyActor."""
```

The fields already have descriptions. Will that be enough to list them in the docs, or do we need to copy those descriptions to the docstring? If so, could you copy them too?

**Comment by can-anyscale** [comment]
@shrekris-anyscale: thanks, will do!

**Review by edoakes — APPROVED** [review]

**Pull request merged**